src/Compilers/CSharp/Portable/FlowAnalysis/DefiniteAssignment.cs

```diff
@@ -160,7 +156,7 @@ internal partial class DefiniteAssignmentPass : LocalDataFlowPass<
                    trackUnassignments)
         {
             this.initiallyAssignedVariables = null;
-            _sourceAssembly = ((object)member == null) ? null : (SourceAssemblySymbol)member.ContainingAssembly;
+            _sourceAssembly = member?.ContainingAssembly as SourceAssemblySymbol;
```

Comment:
What is member in this case? And why is its assembly not from source?

Reply:
The member in the failure case is an attribute where the attribute type is from metadata.

Reply:
Fascinating. Can we add an assert that it's basically either this case, or from source? It feels like this could potentially hide a bug later on if another case slipped in here.